docs/spec/glossary.md

```diff
@@ -0,0 +1,99 @@
+
+**Application Service Mesh** - A Service Mesh implementation primary focusing on payloads which are L4 (TCP streams/UDP datagrams) and L7 (HTTP/Application specific protocols).
+
+### Network Service Mesh definitions
+
+**Network Service Mesh** - A Service Mesh implementation primary focused on carrying payloads which are L2 (e.g. frames) and L3 (e.g. packets).
+
+**Network Service** - A Network Service is a developer-oriented way of looking at how packets receive treatment from the Network.
+
+It is the intersection of:
+ * Connectivity to a set of network-reachable resources
+ * Security
+ * Network functions
+    * Proxying
+    * Load balancing
+    * NAT
+    * QoS
+    * etc.
+
+In NSM, a Network Service is expressed in terms that developers understand:
+ * Nominate the entity you want via a pre-agreed upon name
+ * Select a specific network service by filtering via label metadata that has been applied to available services
+
+### Network Service Mesh elements
+
+**Access control** - A method of ensuring that the selected service is capable of being selective when it accepts a connection
+
+**Workload** - A POD, virtual machine, or running software on a physical device.  A Workload may be a Network Service Client requesting zero or more Network Services from the Network Service Mesh.  A Workload may be a Network Service Endpoint offering zero or more Network Services to the Network Service Mesh.  A Workload may simultaneously be both a Network Service Endpoint and Network Service Client.
+
+**Network-reachable resource** - An element on the network capable of accepting communication: for instance, a software service answering queries.
+ * May be specified as part of a group by choosing a specific network location (e.g. a corporate network), that may be accessed over the connection provided via this network service
+ * This is a primary resource - the network location should be implied perhaps via a constraint, rather than stated as a requirement of the service when possible with the aim being to shield the develop from extraneous networking details.
+
+### Network Service Mesh components
+
+**Network Service Endpoint** - a container, POD, virtual machine or physical forwarder with knowledge of network resources.  A network service endpoint accepts zero or more connection requests from clients which want to receive the Network Service it is offering.
+
+**Network Service Client** - a requester or consumer of the Network Service.
+
+**Network service registry** -  A cluster, VIM or physical network level registry of NSM components
+
+**Network Service Registry Domain** - The registry of all network services, network service endpoint(s) providing the services, and network service managers registered to a specific NSR
+
+**Network Service Manager Domain** - The collective of Network Service Client(s) and Network Service Endpoint(s) directly managed by an NSMgr.
```

Comment:
This should probably come after the definition of Network Service Manager.

Reply:
Don't want to hold this up over a nit... would be good to fix with follow up PR.

Reply:
agreed. This one is on me, the shuffling got a little out of order.